libvast/src/system/infer_command.cpp

```diff
@@ -63,25 +63,29 @@ infer(const std::string& input, const caf::settings& options) {
   return result;
 }
 
-type deduce(const json& j) {
-  using namespace vast;
-  auto f = detail::overload{
-    [](json::null) { return type{}; },
-    [](json::boolean) -> type { return bool_type{}; },
-    [](json::number x) -> type {
-      // TODO: we should include the string representation of the value to make
-      // a good guess because at this point we no longer know whether the input
-      // was "0" or "0.0".
-      json::number i;
-      if (x == 0.0 || std::modf(x, &i) != 0)
-        return real_type{};
-      if (x < 0)
-        return integer_type{};
+type deduce(simdjson::dom::element e) {
```

Comment:
There is a ambiguity in how to interpret a given json values as types.
This is example from integration test (which should fail currently):

```
{ "a": null, "b": true, "c": false, "d": 0, "e": -0, "f": -42, "g": 4.2, "h": 42.0, "i": 42}
-type json = record{a: none, b: bool, c: bool, d: real, e: real, f: int, g: real, h: count, i: count}
+type json = record{a: none, b: bool, c: bool, d: int, e: int, f: int, g: real, h: real, i: int}
```

Reply:
That's just the nature of `vast::data` being more powerful than JSON. I'll discuss with the team which conversions are preferred.

Reply:
After looking at the source code some more, I think the way you implemented this is absolutely correct. Please update the integration test accordingly.